host-spec/c06-consensus.tm

```diff
@@ -505,41 +610,35 @@
 
   <subsection|Epoch Randomness><label|sect-epoch-randomness>
 
-  At the end of epoch <math|\<cal-E\><rsub|n>>, each block producer is able
-  to compute the randomness seed it needs in order to participate in the
-  block production lottery in epoch <math|\<cal-E\><rsub|n+2>>. For epoch 0
-  and 1, the randomness seed is provided in the genesis state. The
-  computation of the seed is described in Algorithm
-  <reference|algo-epoch-randomness> which uses the concept of epoch subchain
-  described in Definition <reference|defn-epoch-subchain>.
+  In the beginning of each epoch <math|\<cal-E\><rsub|n>> the host will
+  receives the randomness seed <math|\<cal-R\><rsub|\<cal-E\><rsub|n+1>>>(def.
+  <reference|defn-epoch-randomness>) necessary to participate in the block
+  production lottery in the next epoch <math|\<cal-E\><rsub|n+1>> from the
+  runtime, through a consesus message in the digest (def.
+  <reference|defn-consensus-message-digest>) of the first block.
 
-  <\algorithm>
-    <label|algo-epoch-randomness><name|Epoch-Randomness>(<math|n\<gtr\>2:>epoch
-    index)
-  <|algorithm>
-    <\algorithmic>
-      <\state>
-        <math|\<rho\>\<leftarrow\>\<phi\>>
-      </state>
+  <\definition>
+    <label|defn-epoch-randomness>For epoch <math|\<cal-E\>> there is a 32-byte
+    <with|font-series|bold|randomness seed> <math|\<cal-R\><rsub|\<cal-E\>>>
+    computed based on the previous epochs VRF outputs. For
+    <math|\<cal-E\><rsub|0>> and <math|\<cal-E\><rsub|1>>, the randomness
+    seed is provided in the genesis state.
+  </definition>
 
-      <\state>
-        <FOR-IN|<math|B>|><em|<name|SubChain(<math|\<cal-E\><rsub|n-2>>)>>
-      </state>
+  \;
 
-      <\state>
-        <math|\<rho\>\<leftarrow\>\<rho\><around*|\|||\|>d<rsub|B>><END>
-      </state>
+  <\definition>
```

Comment:
I did not understand here.

Reply:
This was moved to the runtime specification, as the randomness is now computed in the runtime.